src/main/scala/shine/DPIA/Data.scala

```diff
@@ -14,6 +14,8 @@ final case class IntData(i: Int) extends Data(int) {
   override def toString: String = i.toString
 }
 
+final case class NatAsIntData(n: Nat) extends Data(int)
```

Comment:
Is there a reason to introduce this change in this PR, or is it orthogonal? Does this not require changes to `VisitAndRebuild` and others? Maybe we need to think about cleaning up `Literal(NatData(_))`, `Natural(_)` and other constructs at some point.

Reply:
This and your next comment are linked:

adding the explicit type information to `KernelCallCmd` has exposed this "bug" that the type we record (`exp[int, read]`) and the argument value we create (a `Natural`) did not match.
This was not a problem before, as (as you say in the commend below) "is going to be the same at runtime".

To bring the types and arguments into line, I added this way to add a literal with a natural number whose type is an int. This is obviously not ideal.

I will see if I can make the opposite change (i.e. change the type rather than the value), but I believe that had greater implications elsewhere ...

I agree that we should at some point revisit our `Literal`s, `Natural`, etc.

Reply:
I created an issue to track this: #172